src/modm/platform/core/cortex/build_id.cpp

```diff
@@ -15,19 +15,17 @@ typedef struct {
     uint32_t namesz;
     uint32_t descsz;
     uint32_t type;
-    uint8_t data[];
+    uint8_t data[30];
 } ElfNoteSection_t;
-extern "C" const ElfNoteSection_t __build_id[];
+extern "C" const ElfNoteSection_t __build_id;
```

Comment:
Does this work in hardware? I found that the [] is necessary to get the right pointer.

Reply:
Yes, it's tested, you can do both. You only need a pointer if the struct is an array.